orchagent/portsorch.cpp

```diff
@@ -691,24 +782,43 @@ void PortsOrch::doVlanTask(Consumer &consumer)
 
         if (op == SET_COMMAND)
         {
-            /* Duplicate entry */
-            if (m_portList.find(vlan_alias) != m_portList.end())
+            if (m_portList.find(vlan_alias) == m_portList.end())
             {
-                it = consumer.m_toSync.erase(it);
-                continue;
+                if (!addVlan(vlan_alias))
+                {
+                    it++;
+                    continue;
+                }
             }
 
-            if (addVlan(vlan_alias))
-                it = consumer.m_toSync.erase(it);
-            else
-                it++;
+            uint32_t mtu = 0;
+            for (auto i : kfvFieldsValues(t))
+            {
+                /* Set port mtu */
+                if (fvField(i) == "mtu")
+                {
+                    mtu = (uint32_t)stoul(fvValue(i));
+                }
+
+                // TODO: UNKNOWN_UNICAST_FLOOD_GROUP, UNKNOWN_MULTICAST_FLOOD_GROUP, BROADCAST_FLOOD_GROUP
+            }
+
+            if (mtu != 0)
+            {
+                Port v;
+                getPort(vlan_alias, v);
+                v.m_mtu = mtu;
+                // TODO: update MTU for the IP interface of the VLAN
+            }
```

Comment:
since it is todo, I suggest to remove it. The rif MTU should be configured under interface_table.

Reply:
done.  
btw,  VLAN_TABLE schema might needs some update to remove mtu field.
mtu                 = 1*4DIGIT             ; MTU for the IP interface of the VLAN